### DISCUSSION TITLE
Flexible response body based on user input of list of fields as a query parameter.

### DISCUSSION THREAD

**Discussion by parimalp-bi** [fastapi/fastapi#8899] [Questions]
```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/blogs/?fieldset=creator,name")
def read_root(listOfFields=fieldset):
'''
return response which should only contain two fields but blogs resource may contain 100 fields.    

```

### Description

I have a say two tables blogs and users. One blog can have multiple creators. So its one to many relationship.
To achieve above functionality how should we go ahead ?

1. Is it possible with FastAPIs  pydantic response schema ? Can schema be created runtime based on query parameter ?
2. Or We should create a query based on query parameter and create a response instead of using response schema ?
3. If second option is good can we use SQL alchemy or use raw sqls ? 

### Environment

* OS: [Windows]:
* FastAPI Version [0.65.2]:
* Python version:3.9.0

**Comment by sathoune** [comment]
Do you mean something like this:

```python
from typing import Set
from pydantic import BaseModel


class BigTable(BaseModel):
    name: str
    age: int
    something_else: str


item = BigTable(
    name="Michael",
    age=66,
    something_else="mamma mia",
)


def serialize_and_include(instance: BigTable, to_include: Set):
    return instance.json(include=to_include)


print(serialize_and_include(item, {"age", "something_else"}))
print(serialize_and_include(item, {"name"}))
```

That prints for me:
```bash
{"age": 66, "something_else": "mamma mia"}
{"name": "Michael"}
```

You could do the opposite with `exclude` on `.json`, `.dict` method should have the same parameters.

**Comment by sathoune** [comment]
I have a working app for you expanding on the above example:

```python
from typing import Set

import fastapi
from pydantic import BaseModel


class BigTable(BaseModel):
    name: str
    age: int
    something_else: str


class FalsyBigTable(BigTable):
    name: str = None
    age: int = None
    something_else: str = None

item = BigTable(
    name="Michael",
    age=66,
    something_else="mamma mia",
)


def serialize_and_include(instance: BigTable, to_include: Set):
    return instance.dict(include=to_include)


app = fastapi.FastAPI()


@app.get(
    "/fields",
    response_model=FalsyBigTable,
    response_model_exclude_none=True
)
def get_fields(include: str = fastapi.Query(...)):
    parsed_query = set(include.split(','))
    return serialize_and_include(item, parsed_query)
```
run with
```bash
uvicorn main:app --reload
```
You should be able to query the specific fields with keys separated with `,`. 

I included a response model with None values so you can get validation and documentation generated. You could skip `response_model` and `response_model_exclude_none` if you don't need that.

**Comment by parimalp-bi** [comment]
in this data is being fetched fully from DB .. i.e. all fields .. I don't want that to happen. 
I want to show full response that be return by api endpoint but at the same time only fetch columns what is given as parameters by user

**Reply by YuriiMotov** [reply]
You can construct SQL statement according to input parameters. This way you can fetch only fields that are needed
